pilot/pkg/model/push_context.go

```diff
@@ -1449,7 +1450,18 @@ func (ps *PushContext) initServiceRegistry(env *Environment, configsUpdate sets.
 		if _, f := ps.ServiceIndex.HostnameAndNamespace[s.Hostname]; !f {
 			ps.ServiceIndex.HostnameAndNamespace[s.Hostname] = map[string]*Service{}
 		}
-		ps.ServiceIndex.HostnameAndNamespace[s.Hostname][s.Attributes.Namespace] = s
+		// In some scenarios, there may be multiple Services defined for the same hostname due to ServiceEntry allowing
```

Comment:
same hostname AND same namespace? :nauseated_face:

Reply:
yeah for SE